misk-redis/src/main/kotlin/misk/redis/Redis.kt

```diff
@@ -721,3 +760,140 @@ inline fun checkHrandFieldCount(count: Long) {
     "You must request at least 1 field."
   }
 }
+
+/**
+ * Like [Redis], but returns [Supplier]s to defer value retrieval.
+ * **Does not support transactions or pubsub.**
+ */
+interface DeferredRedis {
+  fun del(key: String): Supplier<Boolean>
+
+  fun del(vararg keys: String): Supplier<Int>
+
+  fun mget(vararg keys: String): Supplier<List<ByteString?>>
+
+  fun mset(vararg keyValues: ByteString): Supplier<Unit>
+
+  operator fun get(key: String): Supplier<ByteString?>
+
+  fun hdel(key: String, vararg fields: String): Supplier<Long>
+
+  fun hget(key: String, field: String): Supplier<ByteString?>
+
+  fun hgetAll(key: String): Supplier<Map<String, ByteString>?>
+
+  fun hlen(key: String): Supplier<Long>
+
+  fun hmget(key: String, vararg fields: String): Supplier<List<ByteString?>>
+
+  fun hincrBy(key: String, field: String, increment: Long): Supplier<Long>
+
+  fun hrandFieldWithValues(key: String, count: Long): Supplier<Map<String, ByteString>?>
+
+  fun hrandField(key: String, count: Long): Supplier<List<String>>
+
+  fun set(key: String, value: ByteString, expiryDuration: Duration? = null): Supplier<Unit>
+
+  fun setnx(key: String, value: ByteString, expiryDuration: Duration? = null): Supplier<Boolean>
```

Comment:
Note: set/setnx are intentionally and slightly different from the main Redis API. 

`set` is not an operator, because it is not ergonomic to use it as an operator within a `pipelining` block.

Expiry dates are always optional with `set`, so they are moved to be the last argument with a default of no expiry.